src/Symfony/Component/Debug/BootstrappingLogger.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Debug;
+
+use Psr\Log\AbstractLogger;
+use Psr\Log\LoggerInterface;
+
+/**
+ * A bootstrapping logger that stacks logs and flushes them when required to an other logger.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class BootstrappingLogger extends AbstractLogger
```

Comment:
well, the logger itself is not aware of this behavior. You could reuse it for what you want